### PR TITLE
feature/20624-treemap-drill-to-last-leaf

### PR DESCRIPTION
**Pull request by hubertkozik**
Added new option `treemap.traverseToLeaf` which allows to drill into last possible leaf of Treemap series.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206630380830692

### CODE REVIEW

File: samples/highcharts/plotoptions/treemap-zoomtolastleaf/demo.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+Highcharts.chart('container', {
+    series: [{
+        type: 'treemap',
+        layoutAlgorithm: 'squarified',
+        allowTraversingTree: true,
+        interactByLeaf: true,
+        allowZoomToLastLeaf: true,
```
[the remark this answers is not in the document]
Comment (hubertkozik) [reply samples/highcharts/plotoptions/treemap-zoomtolastleaf/demo.js:7]: Whooops, should be.

File: samples/highcharts/plotoptions/treemap-zoomtolastleaf/demo.details [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+ name: Treemap Demo
+ authors:
+   - Hubert Kozik
+ js_wrap: b
+...
```
[the remark this answers is not in the document]
Comment (hubertkozik) [reply samples/highcharts/plotoptions/treemap-zoomtolastleaf/demo.details]: I've added extra line to every file, but it is not showing up in GH "Files changed", it's in code.

**Review by bre1470 — APPROVED** [review]
Nice! 👍

**Comment by TorsteinHonsi** [comment]
Thanks! All good in the code, but some notes on the naming:

* In the description for the changelog, the name `drillToLastLeaf` is used.
* In the code, it is `zoomToLastLeaf`.
* In the past, we have tried to use the term _traversing_ constantly for drilling into a tree map. This is because it is technically not the same operation as the drilldown of the Drilldown module. We wanted to avoid confusion between the two. While the Drilldown module exchanges a point for a whole new series, the traversing of a treemap just changes the root node of the tree. So this new feature, isn't it really about traversing? Zooming indicates a purely visual operation, that we just change the view without changing the tree itself.

**Comment by hubertkozik** [comment]
@TorsteinHonsi what do you think about changing the name of a new option to `traverseToLastLeaf`? I agree, that the word `zoom` can be wrong.

**Comment by TorsteinHonsi** [comment]
`traverseToLastLeaf` sounds good.

Or, wait a minute - isn't "last leaf" redundant? I mean a leaf is by definition the last node on a tree, right? So `traverseToLeaf` should be sufficient?

File: samples/highcharts/plotoptions/treemap-traverse-to-leaf/demo.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+Highcharts.chart('container', {
+    series: [{
+        type: 'treemap',
+        layoutAlgorithm: 'squarified',
+        allowTraversingTree: true,
+        interactByLeaf: true,
+        zoomToLastLeaf: true,
```
Comment (TorsteinHonsi) [inline samples/highcharts/plotoptions/treemap-traverse-to-leaf/demo.js:7]:
```suggestion
        traverseToLeaf: true,
```

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Except for the hands-on demo not working since the name change, everything looks good!

On a small note, Øystein and I discussed that this option does two things:
* Allows traversing to the leaf
* Changes clicking to go _directly_ to the leaf (skipping intermediate levels)
We agreed that in the future, if someone wants to split this functionality, we can introduce string/enum settings for the option. Something like

`traverseToLeaf: boolean|'directly'|'stepwise'`

**Comment by hubertkozik** [comment]
Whoops, sorry with the option name in the demo, too many changes of name, and here's the result 😁 Do you think we should create a Feature Request ticket with that idea or maybe simply add it to Asana?

**Comment by TorsteinHonsi** [comment]
> About future, do you think we should create a Feature Request ticket with that idea or maybe simply add it to Asana?

I don't think we need to record it anywhere. Let's just wait until someone requests it, most likely they never will.

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**